src/library_manager/lib_manager.c

```diff
@@ -332,26 +333,26 @@ static void lib_manager_update_sof_ctx(struct sof_man_fw_desc *desc, uint32_t li
 
 int lib_manager_register_module(struct sof_man_fw_desc *desc, int module_id)
 {
+	uint32_t entry_index = LIB_MANAGER_GET_MODULE_INDEX(module_id);
 	/* allocate new  comp_driver_info */
 	struct comp_driver_info *new_drv_info;
 	struct comp_driver *drv = NULL;
 	struct sof_man_module *mod;
 	int ret;
 
-	uint32_t align = 4;
-	uint32_t entry_index = LIB_MANAGER_GET_MODULE_INDEX(module_id);
-
-	new_drv_info = rballoc_align(0, SOF_MEM_CAPS_RAM,
-				     sizeof(struct comp_driver_info), align);
+	new_drv_info = rmalloc(SOF_MEM_ZONE_RUNTIME_SHARED, 0,
+			       SOF_MEM_CAPS_RAM | SOF_MEM_FLAG_COHERENT,
+			       sizeof(struct comp_driver_info));
```

Comment:
@lyakh, can you help me out with this?

Reply:
this change does several things:
1. no need to use `rballoc()` - this isn't a data buffer, this is just a normal structure object
2. allocating as "shared" means - uncached. I think that's appropriate because that's what seems to be used for `comp_register()` in other places.